tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+
+# cert-manager install process adapted from:
+# https://docs.cert-manager.io/en/latest/getting-started/install/kubernetes.html#installing-with-regular-manifests
+- name: Create namespace for cert-manager
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    validate:
+      fail_on_error: yes
+      strict: yes
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: cert-manager
+  register: cert_manager
+
+- name: Temporarily disable validation for cert-manager namespace
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    validate:
+      fail_on_error: yes
+      strict: yes
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: cert-manager
+        labels:
+          certmanager.k8s.io/disable-validation: "true"
+  when: cert_manager is changed
+
+- name: Deploy cert-manager and Let's Encrypt
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    # FIXME: cert-manager.yaml doesn't validate, so this is disabled for now.
+    # Maybe the next version will validate?
+    # validate:
+    #   fail_on_error: yes
+    #   strict: yes
+    definition: "{{ lookup('template', item) }}"
+  with_items:
+    - letsencrypt/cert-manager.yaml
+    - letsencrypt/issuers.yaml
+
+# ingress-nginx installation process adapted from:
+# https://kubernetes.github.io/ingress-nginx/deploy/#prerequisite-generic-deployment-command
+- name: Deploy Nginx Ingress Controller
+  k8s:
+    context: "{{ k8s_context }}"
+    kubeconfig: "{{ k8s_kubeconfig }}"
+    wait: yes
+    validate:
+      fail_on_error: yes
+      strict: yes
+    definition: "{{ lookup('template', item) }}"
+  with_items:
+    - ingress-nginx/mandatory.yaml
```

Comment:
It looks as if we've modified the other two yaml files, but if we haven't modified `mandatory.yaml`, we could install from the source URL and not need our own copy (and make it clear we're using it unmodified).  Here's how I'm doing that in another project that is using all three files unmodified:

```  - name: install nginx ingress controller
    # https://kubernetes.github.io/ingress-nginx/deploy/
    k8s:
      definition: "{{ lookup('url', item, split_lines=False) }}"
      state: present
      wait: yes
      validate:
        fail_on_error: yes
        strict: yes
    with_items:
    - https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.29.0/deploy/static/mandatory.yaml
    - https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.29.0/deploy/static/provider/aws/service-l4.yaml
    - https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.29.0/deploy/static/provider/aws/patch-configmap-l4.yaml
```

Reply:
Added to #3